analytics/javaspark/src/main/java/mil/nga/giat/geowave/analytic/javaspark/KMeansRunner.java

```diff
@@ -0,0 +1,225 @@
+package mil.nga.giat.geowave.analytic.javaspark;
+
+import java.io.IOException;
+import java.util.StringTokenizer;
+
+import org.apache.spark.SparkConf;
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.mllib.clustering.KMeans;
+import org.apache.spark.mllib.clustering.KMeansModel;
+import org.apache.spark.mllib.linalg.Vector;
+import org.opengis.feature.simple.SimpleFeature;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.vividsolutions.jts.geom.Envelope;
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.geom.GeometryFactory;
+
+import mil.nga.giat.geowave.core.geotime.store.query.ScaledTemporalRange;
+import mil.nga.giat.geowave.core.geotime.store.query.SpatialQuery;
+import mil.nga.giat.geowave.core.index.StringUtils;
+import mil.nga.giat.geowave.core.store.CloseableIterator;
+import mil.nga.giat.geowave.core.store.adapter.DataAdapter;
+import mil.nga.giat.geowave.core.store.operations.remote.options.DataStorePluginOptions;
+import mil.nga.giat.geowave.core.store.query.DistributableQuery;
+import mil.nga.giat.geowave.core.store.query.QueryOptions;
+import mil.nga.giat.geowave.mapreduce.input.GeoWaveInputKey;
+
+public class KMeansRunner
+{
+	private final static Logger LOGGER = LoggerFactory.getLogger(KMeansRunner.class);
+
+	private String appName = "KMeansRunner";
+	private String master = "local";
+	private String host = "localhost";
+
+	private JavaSparkContext jsc = null;
+	private DataStorePluginOptions inputDataStore = null;
+	private JavaRDD<Vector> centroidVectors;
+	private KMeansModel outputModel;
+
+	private int numClusters = 8;
+	private int numIterations = 20;
+	private double epsilon = -1.0;
+	private Envelope bbox = null;
+	private String adapterId = null;
+	private String timeField = null;
+	private ScaledTemporalRange scaledTimeRange = null;
+
+	public KMeansRunner() {}
+
+	private void initContext() {
+		SparkConf sparkConf = new SparkConf();
+
+		sparkConf.setAppName(appName);
+		sparkConf.setMaster(master);
+
+		// TODO: other context config settings
+
+		jsc = new JavaSparkContext(
+				sparkConf);
+	}
+
+	public void closeContext() {
+		if (jsc != null) {
+			jsc.close();
+			jsc = null;
+		}
+	}
+
+	public void run()
+			throws IOException {
+		initContext();
+
+		// Validate inputs
+		if (inputDataStore == null) {
+			LOGGER.error("You must supply an input datastore!");
+			throw new IOException(
+					"You must supply an input datastore!");
+		}
+
+		QueryOptions queryOptions = null;
+		if (adapterId != null) {
+			// Retrieve the adapters
+			CloseableIterator<DataAdapter<?>> adapterIt = inputDataStore.createAdapterStore().getAdapters();
```

Comment:
you can just setAdapterId() on queryoptions and not need to iterate through adapters to find a match